### DISCUSSION TITLE
Tips for using ADC?

### DISCUSSION THREAD

**Discussion by grandaspanna** [SpenceKonde/DxCore#234] [Q&A]
Maybe there's a better place to ask, so I'm happy to be told where to go...

I've implemented a design where one of its functions is to monitor battery voltage. Overall, it works well - I have a simple resistor divider and using VCC as reference (currently). I'm calling analogRead() about every 20ms, but every now and then, I get a single reading that is quite "spiked":

7.68
7.69
**8.25**
7.66
7.67
7.64
7.69

There's no apparent fixed period or cycle to it (the blue line):
![image](https://user-images.githubusercontent.com/54263749/150090768-a66a93d2-15a5-4fe6-8113-3998e509551f.png)

So far, it's easy enough to smooth in analysis, but I'm wondering if there are smarter ways to use the ADC. I'm aiming for around 50-100 samples a second in order to track dips under load.

Any tips?

**Comment by SpenceKonde** [comment]
First I've heard of any phenomenon like that... I'd check for a possible wiring issue (bad connection on one side of resistor divider?)

**Reply by grandaspanna** [reply]
I'll check that. The resistors are SMD, but yeah. not guarantee they or the chip pad are 100% with my soldering...

I wasn't sure if playing with any of the sampling/bursting features was worth it. My first foray into analog :-)

**Comment by SpenceKonde** [comment]
You could certainly try taking samples with analogReadEnh() (seethe analog documentation supplied with the core) and either specifying a higher resolution (which will oversample and decimate) or simply using accumulation and then dividing it yourself - but if there's a problem with the electrical portion of it, you want that sorted out, even if you could hide it :-P

**Reply by grandaspanna** [reply]
Thanks. I've read over the analog doco a few times and each time I think I learn a little more.
